File: Data Structures & Algorithms/custom-sort-string/submission-0.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        vector<int> freq(26, 0);

        // Step 1: Count frequency
        for(char c : s) {
            freq[c - 'a']++;
        }

        string res;

        // Step 2: Add characters in given order
        for(char c : order) {
            while(freq[c - 'a'] > 0) {
                res.push_back(c);
                freq[c - 'a']--;   // 🔥 important
            }
        }

        // Step 3: Add remaining characters
        for(int i = 0; i < 26; i++) {
            while(freq[i] > 0) {
                res.push_back(i + 'a');
                freq[i]--;   // 🔥 important
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/merge-strings-alternately/submission-0.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        n1 = len(word1)
        n2 = len(word2)
        i = 0
        res = ""

        while i<n1 and i<n2:
            res += word1[i] + word2[i]
            i+=1

        if i==n1:
            res += word2[i:]
        if i==n2:
            res += word1[i:]
        
        return res
